src/ethereum_test_tools/common/types.py

```diff
@@ -995,6 +1014,102 @@ def to_list(self) -> List[Address | List[Hash]]:
         return [self.address, self.storage_keys]
 
 
+class AuthorizationTupleGeneric(CamelModel, Generic[NumberBoundTypeVar]):
+    """
+    Authorization tuple for transactions.
+    """
+
+    chain_id: NumberBoundTypeVar = Field(1)  # type: ignore
+    address: Address
+    nonce: List[NumberBoundTypeVar] = Field(default_factory=list)
```

Comment:
the nonce is an optional type (the eip presents it as `[Nonce]`, which might be confusing) and not a list type.